lib/rules/unicode-bom/index.js

```diff
@@ -0,0 +1,54 @@
+"use strict";
+
+const report = require("../../utils/report");
+const ruleMessages = require("../../utils/ruleMessages");
+const validateOptions = require("../../utils/validateOptions");
+
+const ruleName = "unicode-bom";
+
+const messages = ruleMessages(ruleName, {
+  expected: "Expected Unicode BOM",
+  rejected: "Unexpected Unicode BOM"
+});
+
+const rule = function(expectation) {
+  return (root, result) => {
+    const validOptions = validateOptions(result, ruleName, {
+      actual: expectation,
+      possible: ["always", "never"]
+    });
+
+    if (
+      !validOptions ||
+      root.source.inline ||
+      root.source.lang === "object-literal" ||
+      root.document !== undefined
+    ) {
+      return;
+    }
+
+    const { hasBOM } = root.source.input;
+
+    if (expectation === "always" && !hasBOM) {
+      report({
+        result,
+        ruleName,
+        message: messages.expected,
+        root,
+        line: 1
+      });
+    } else if (expectation === "never" && hasBOM) {
```

Comment:
```suggestion
    }

    if (expectation === "never" && hasBOM) {
```

It will increase readability.